tee-worker/litentry/core/data-providers/src/achainable.rs

```diff
@@ -155,196 +92,428 @@ pub trait AchainableResultParser {
+	}
+}
+
+impl ReqBody {
+	pub fn new(address: String, params: Params) -> Self {
+		ReqBody { name: params.name(), address, params }
+	}
+}
+
+pub trait AchainableSystemLabelName {
+	fn name(&self) -> String;
+}
+
+pub fn web3_network_to_chain(network: &Web3Network) -> String {
+	match network {
+		Web3Network::Polkadot => "polkadot".into(),
+		Web3Network::Kusama => "kusama".into(),
+		Web3Network::Litentry => "litentry".into(),
+		Web3Network::Litmus => "litmus".into(),
+		Web3Network::LitentryRococo => "litentry_rococo".into(),
+		Web3Network::Khala => "khala".into(),
+		Web3Network::SubstrateTestnet => "substrate_testnet".into(),
+		Web3Network::Ethereum => "ethereum".into(),
+		Web3Network::Polygon => "polygon".into(),
+		Web3Network::BSC => "bsc".into(),
+	}
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(untagged)]
+pub enum Params {
+	ParamsBasicType(ParamsBasicType),
+	ParamsBasicTypeWithAmount(ParamsBasicTypeWithAmount),
+	ParamsBasicTypeWithAmounts(ParamsBasicTypeWithAmounts),
+	ParamsBasicTypeWithDate(ParamsBasicTypeWithDate),
+	ParamsBasicTypeWithAmountToken(ParamsBasicTypeWithAmountToken),
+	ParamsBasicTypeWithBetweenPercents(ParamsBasicTypeWithBetweenPercents),
+	ParamsBasicTypeWithDateInterval(ParamsBasicTypeWithDateInterval),
+	ParamsBasicTypeWithToken(ParamsBasicTypeWithToken),
+	ParamsBasicTypeWithDatePercent(ParamsBasicTypeWithDatePercent),
+	ParamsBasicTypeWithClassOfYear(ParamsBasicTypeWithClassOfYear),
+	ParamsBasicTypeWithAmountHoding(ParamsBasicTypeWithAmountHoding),
+}
+
+impl AchainableSystemLabelName for Params {
+	fn name(&self) -> String {
+		match self {
+			Params::ParamsBasicType(a) => a.name.clone(),
+			Params::ParamsBasicTypeWithAmount(a) => a.name.clone(),
+			Params::ParamsBasicTypeWithAmounts(a) => a.name.clone(),
+			Params::ParamsBasicTypeWithDate(a) => a.name.clone(),
+			Params::ParamsBasicTypeWithAmountToken(a) => a.name.clone(),
+			Params::ParamsBasicTypeWithBetweenPercents(a) => a.name.clone(),
+			Params::ParamsBasicTypeWithDateInterval(a) => a.name.clone(),
+			Params::ParamsBasicTypeWithToken(a) => a.name.clone(),
+			Params::ParamsBasicTypeWithDatePercent(e) => e.name.clone(),
+			Params::ParamsBasicTypeWithClassOfYear(c) => c.name.clone(),
+			Params::ParamsBasicTypeWithAmountHoding(a) => a.name.clone(),
 		}
+	}
+}
 
-		let path = if *network == Web3Network::Ethereum {
-			A4_ERC20_LIT_ETHEREUM_PATHS[index]
-		} else if *network == Web3Network::Litentry {
-			A4_LIT_LITENTRY_PATHS[index]
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct ParamsBasicTypeWithAmountHoding {
+	#[serde(skip_serializing)]
+	#[serde(skip_deserializing)]
+	pub name: String,
+
+	pub chain: String,
+	pub amount: String,
+	pub date: String,
+
+	#[serde(skip_serializing_if = "Option::is_none")]
+	pub token: Option<String>,
+}
+
+impl ParamsBasicTypeWithAmountHoding {
+	pub fn new(netwok: &Web3Network, amount: String, date: String, token: Option<String>) -> Self {
```

Comment:
:nail_care: just spelling:

```suggestion
impl ParamsBasicTypeWithAmountHolding {
	pub fn new(network: &Web3Network, amount: String, date: String, token: Option<String>) -> Self {
```